mysql-test/main/mysql_json_mysql_upgrade_system_tables.test

```diff
@@ -0,0 +1,52 @@
+--echo #
+--echo # MDEV-32462: mysql_upgrade -s still checks for non system tables
+--echo #
+
+# Let's now load plugin first
+--source include/have_utf8.inc
+--source include/not_embedded.inc
+
+--source include/mysql_upgrade_preparation.inc
+call mtr.add_suppression("Table rebuild required");
+
+SET NAMES utf8;
+
+let $MYSQLD_DATADIR= `select @@datadir`;
+
+--echo # mariadb_upgrade on system and user table
+--copy_file std_data/mysql_json/mysql_json_test.frm $MYSQLD_DATADIR/mysql/mysql_json_test.frm
+--copy_file std_data/mysql_json/mysql_json_test.MYI $MYSQLD_DATADIR/mysql/mysql_json_test.MYI
+--copy_file std_data/mysql_json/mysql_json_test.MYD $MYSQLD_DATADIR/mysql/mysql_json_test.MYD
+--copy_file std_data/mysql_json/mysql_json_test.frm $MYSQLD_DATADIR/test/mysql_json_test.frm
+--copy_file std_data/mysql_json/mysql_json_test.MYI $MYSQLD_DATADIR/test/mysql_json_test.MYI
+--copy_file std_data/mysql_json/mysql_json_test.MYD $MYSQLD_DATADIR/test/mysql_json_test.MYD
+
+show tables from mysql like '%json%';
+use mysql;
+--error ER_UNKNOWN_DATA_TYPE
+show create table mysql.mysql_json_test;
+--error ER_UNKNOWN_DATA_TYPE
+show create table test.mysql_json_test;
+
+SET @old_general_log= @@global.general_log;
+SET @old_log_output= @@global.log_output;
+SET @@global.general_log = ON;
+SET @@global.log_output = "TABLE";
+--exec $MYSQL_UPGRADE -s --force 2>&1
+--remove_file $MYSQLD_DATADIR/mysql_upgrade_info
+SET @@global.general_log = @old_general_log;
+SET @@global.log_output = @old_log_output;
+
+select command_type, argument from mysql.general_log where argument like "%SELECT table_comment FROM information_schema.tables%";
+
+# User table is not upgraded in `mysql\test` DB, so we cannot see it.
+--error ER_UNKNOWN_DATA_TYPE
+show create table mysql.mysql_json_test;
```

Comment:
This is not good then. I thought that we'd be upgrading mysql database tables after loading the MYSQL_JSON plugin.

If mysql_upgrade is now run with -s parameter, it leads to this current logic that makes no sense:

1. Try to check all tables in mysql database and upgrade them. This will fail for any tables with MYSQL_JSON datatype.
2. Check if we need to load MYSQL_JSON plugin *for* any tables in mysql database.
3. Don't do anything else.

Observing this, let's just skip loading MYSQL_JSON for mysql database, like you @an3l proposed initially,
```c++
static int install_used_plugin_data_types(void)
{
....
if (opt_systables_only)
  return 0;
....
}
```

There's no point in actually going through all that logic, loading the plugin, unloading it, to actually not do anything in the end.

To make sure, I checked all MySQL's tables in 5.7:

```sql
mysql> select distinct(data_type) from information_schema.columns where table_schema='mysql' order by 1;
+------------+
| data_type  |
+------------+
| bigint     |
| blob       |
| char       |
| datetime   |
| enum       |
| float      |
| int        |
| longblob   |
| mediumblob |
| mediumtext |
| set        |
| smallint   |
| text       |
| time       |
| timestamp  |
| tinyint    |
| varchar    |
+------------+
```

So we should be safe ignoring systables in this case.